symengine/ruby/ext/symengine/ruby_basic.c

```diff
@@ -19,12 +19,15 @@ VALUE cbasic_binary_op(VALUE self, VALUE operand2, void (*cwfunc_ptr)(basic_stru
     basic_struct *this, *cbasic_operand2, *cresult;
     VALUE result;
 
+    cbasic_operand2 = basic_new_heap();
+
     Data_Get_Struct(self, basic_struct, this);
-    Data_Get_Struct(operand2, basic_struct, cbasic_operand2);
+    sympify(operand2, cbasic_operand2);
 
     cresult = basic_new_heap();
     cwfunc_ptr(cresult, this, cbasic_operand2);
-    result = Data_Wrap_Struct(rb_obj_class(self), NULL , cbasic_free_heap, cresult);
+    result = Data_Wrap_Struct(Klass_of_Basic(cresult), NULL , cbasic_free_heap, cresult);
+    basic_free_heap(cbasic_operand2);
```

Comment:
I am worried that you are double freeing it here, aren't you? You free the `cbasic_operand2` using `basic_free_heap()` here, and then Ruby will double free it using `cbasic_free_heap`. Or am I misunderstanding it?